### ISSUE TITLE
Action fails with "`add-path` is deprecated" when using self-hosted runner

### ISSUE THREAD

**Issue by chipbuster** [ATiltedTree/setup-rust#64]
Hey there! I'm not 100% certain if this is a configurational error on my end, but I figured I'd ask and see if there was a way to fix it from within this repo.

#### Current Behavior

I have a self-hosted Github Actions runner (v2.285.0) on Rocky Linux 8.5. 

Attempting to use this action on this runner fails with the following messages:

> Unable to process command '::add-path::/home/ghactions/.cargo/bin' successfully.
> The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

After this, the action fails, meaning that my benchmarks are never run. This error does not appear when using GitHub's runners, making me uncertain if this is a configuration error on my end or whether there's some difference between the two environments. (However, I cannot easily run benchmarks on GitHub's runners since the performance data is noisy).

#### Additional context/Screenshots

<details>
  <summary>Complete Github Actions Log</summary>

```
2021-12-07T03:31:52.1027446Z Found online and idle self-hosted runner in the current repository that matches the required labels: 'self-hosted , benchmark'
2021-12-07T03:31:52.1248512Z Waiting for the runner to pick up this job...
2021-12-07T03:31:52.3653076Z Job is about to start running on the runner: ghastly (repository)
2021-12-07T03:31:56.6039859Z Current runner version: '2.285.0'
2021-12-07T03:31:56.6044879Z Runner name: 'ghastly'
2021-12-07T03:31:56.6045740Z Runner group name: 'Default'
2021-12-07T03:31:56.6047132Z Machine name: 'ghastly'
2021-12-07T03:31:56.6051079Z ##[group]GITHUB_TOKEN Permissions
2021-12-07T03:31:56.6052598Z Actions: write
2021-12-07T03:31:56.6053107Z Checks: write
2021-12-07T03:31:56.6053663Z Contents: write
2021-12-07T03:31:56.6054191Z Deployments: write
2021-12-07T03:31:56.6054890Z Discussions: write
2021-12-07T03:31:56.6055463Z Issues: write
2021-12-07T03:31:56.6055951Z Metadata: read
2021-12-07T03:31:56.6056511Z Packages: write
2021-12-07T03:31:56.6057014Z Pages: write
2021-12-07T03:31:56.6057571Z PullRequests: write
2021-12-07T03:31:56.6058188Z RepositoryProjects: write
2021-12-07T03:31:56.6058895Z SecurityEvents: write
2021-12-07T03:31:56.6059439Z Statuses: write
2021-12-07T03:31:56.6060132Z ##[endgroup]
2021-12-07T03:31:56.6063741Z Secret source: Actions
2021-12-07T03:31:56.6065097Z Prepare workflow directory
2021-12-07T03:31:56.6779586Z Prepare all required actions
2021-12-07T03:31:56.6790824Z Getting action download info
2021-12-07T03:31:56.9373962Z Download action repository 'actions/checkout@v2' (SHA:ec3a7ce113134d7a93b817d10a8272cb61118579)
2021-12-07T03:31:58.5887785Z Download action repository 'ATiltedTree/setup-rust@v1' (SHA:e94eb0c2d345d07cc59ad496f20a034bb13a97f6)
2021-12-07T03:32:01.1147258Z ##[group]Run actions/checkout@v2
2021-12-07T03:32:01.1147948Z with:
2021-12-07T03:32:01.1148427Z   repository: chipbuster/fast-hampath
2021-12-07T03:32:01.1149420Z   token: ***
2021-12-07T03:32:01.1149801Z   ssh-strict: true
2021-12-07T03:32:01.1150293Z   persist-credentials: true
2021-12-07T03:32:01.1150755Z   clean: true
2021-12-07T03:32:01.1151134Z   fetch-depth: 1
2021-12-07T03:32:01.1151501Z   lfs: false
2021-12-07T03:32:01.1151878Z   submodules: false
2021-12-07T03:32:01.1152268Z ##[endgroup]
2021-12-07T03:32:02.0154133Z Syncing repository: chipbuster/fast-hampath
2021-12-07T03:32:02.0162769Z ##[group]Getting Git version info
2021-12-07T03:32:02.0184047Z Working directory is '/home/ghactions/actions-runner/_work/fast-hampath/fast-hampath'
2021-12-07T03:32:02.0187443Z [command]/usr/bin/git version
2021-12-07T03:32:02.0187997Z git version 2.27.0
2021-12-07T03:32:02.0190460Z ##[endgroup]
2021-12-07T03:32:02.0191592Z Deleting the contents of '/home/ghactions/actions-runner/_work/fast-hampath/fast-hampath'
2021-12-07T03:32:02.0194008Z ##[group]Initializing the repository
2021-12-07T03:32:02.0195183Z [command]/usr/bin/git init /home/ghactions/actions-runner/_work/fast-hampath/fast-hampath
2021-12-07T03:32:02.0196552Z Initialized empty Git repository in /home/ghactions/actions-runner/_work/fast-hampath/fast-hampath/.git/
2021-12-07T03:32:02.0197955Z [command]/usr/bin/git remote add origin https://github.com/chipbuster/fast-hampath
2021-12-07T03:32:02.0199247Z ##[endgroup]
2021-12-07T03:32:02.0200273Z ##[group]Disabling automatic garbage collection
2021-12-07T03:32:02.0201178Z [command]/usr/bin/git config --local gc.auto 0
2021-12-07T03:32:02.0202184Z ##[endgroup]
2021-12-07T03:32:02.0205060Z ##[group]Setting up auth
2021-12-07T03:32:02.0206052Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2021-12-07T03:32:02.0207562Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2021-12-07T03:32:02.0209099Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2021-12-07T03:32:02.0210983Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2021-12-07T03:32:02.0212835Z [command]/usr/bin/git config --local http.https://github.com/.extraheader AUTHORIZATION: basic ***
2021-12-07T03:32:02.0214122Z ##[endgroup]
2021-12-07T03:32:02.0215108Z ##[group]Fetching the repository
2021-12-07T03:32:02.0216979Z [command]/usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +f7c91ad552d901529ec9ca690dc93f48eef2af78:refs/remotes/origin/devel
2021-12-07T03:32:02.3119157Z remote: Enumerating objects: 17, done.        
2021-12-07T03:32:02.3120757Z remote: Counting objects:   5% (1/17)        
2021-12-07T03:32:02.3121701Z remote: Counting objects:  11% (2/17)        
2021-12-07T03:32:02.3122424Z remote: Counting objects:  17% (3/17)        
2021-12-07T03:32:02.3123338Z remote: Counting objects:  23% (4/17)        
2021-12-07T03:32:02.3123923Z remote: Counting objects:  29% (5/17)        
2021-12-07T03:32:02.3124672Z remote: Counting objects:  35% (6/17)        
2021-12-07T03:32:02.3125183Z remote: Counting objects:  41% (7/17)        
2021-12-07T03:32:02.3125690Z remote: Counting objects:  47% (8/17)        
2021-12-07T03:32:02.3126189Z remote: Counting objects:  52% (9/17)        
2021-12-07T03:32:02.3126701Z remote: Counting objects:  58% (10/17)        
2021-12-07T03:32:02.3127239Z remote: Counting objects:  64% (11/17)        
2021-12-07T03:32:02.3127743Z remote: Counting objects:  70% (12/17)        
2021-12-07T03:32:02.3128249Z remote: Counting objects:  76% (13/17)        
2021-12-07T03:32:02.3128753Z remote: Counting objects:  82% (14/17)        
2021-12-07T03:32:02.3129574Z remote: Counting objects:  88% (15/17)        
2021-12-07T03:32:02.3130116Z remote: Counting objects:  94% (16/17)        
2021-12-07T03:32:02.3130617Z remote: Counting objects: 100% (17/17)        
2021-12-07T03:32:02.3131143Z remote: Counting objects: 100% (17/17), done.        
2021-12-07T03:32:02.3131697Z remote: Compressing objects:   7% (1/13)        
2021-12-07T03:32:02.3132256Z remote: Compressing objects:  15% (2/13)        
2021-12-07T03:32:02.3132809Z remote: Compressing objects:  23% (3/13)        
2021-12-07T03:32:02.3133363Z remote: Compressing objects:  30% (4/13)        
2021-12-07T03:32:02.3133917Z remote: Compressing objects:  38% (5/13)        
2021-12-07T03:32:02.3134544Z remote: Compressing objects:  46% (6/13)        
2021-12-07T03:32:02.3135103Z remote: Compressing objects:  53% (7/13)        
2021-12-07T03:32:02.3135648Z remote: Compressing objects:  61% (8/13)        
2021-12-07T03:32:02.3136199Z remote: Compressing objects:  69% (9/13)        
2021-12-07T03:32:02.3502194Z remote: Compressing objects:  76% (10/13)        
2021-12-07T03:32:02.3503163Z remote: Compressing objects:  84% (11/13)        
2021-12-07T03:32:02.3503962Z remote: Compressing objects:  92% (12/13)        
2021-12-07T03:32:02.3504878Z remote: Compressing objects: 100% (13/13)        
2021-12-07T03:32:02.3505661Z remote: Compressing objects: 100% (13/13), done.        
2021-12-07T03:32:02.3515223Z remote: Total 17 (delta 0), reused 11 (delta 0), pack-reused 0        
2021-12-07T03:32:02.3612674Z From https://github.com/chipbuster/fast-hampath
2021-12-07T03:32:02.3614585Z  * [new ref]         f7c91ad552d901529ec9ca690dc93f48eef2af78 -> origin/devel
2021-12-07T03:32:02.3637502Z ##[endgroup]
2021-12-07T03:32:02.3639307Z ##[group]Determining the checkout info
2021-12-07T03:32:02.3640674Z ##[endgroup]
2021-12-07T03:32:02.3641872Z ##[group]Checking out the ref
2021-12-07T03:32:02.3643485Z [command]/usr/bin/git checkout --progress --force -B devel refs/remotes/origin/devel
2021-12-07T03:32:02.3713679Z Switched to a new branch 'devel'
2021-12-07T03:32:02.3726075Z Branch 'devel' set up to track remote branch 'devel' from 'origin'.
2021-12-07T03:32:02.3727724Z ##[endgroup]
2021-12-07T03:32:02.3761076Z [command]/usr/bin/git log -1 --format='%H'
2021-12-07T03:32:02.3816347Z 'f7c91ad552d901529ec9ca690dc93f48eef2af78'
2021-12-07T03:32:02.3990899Z ##[group]Run ATiltedTree/setup-rust@v1
2021-12-07T03:32:02.3991422Z with:
2021-12-07T03:32:02.3991815Z   rust-version: stable
2021-12-07T03:32:02.3992223Z ##[endgroup]
2021-12-07T03:32:02.7186676Z [command]/home/ghactions/actions-runner/_work/_temp/316d7065-9c56-4c5c-b21f-0fdfb9f483e2 --default-toolchain none -y
2021-12-07T03:32:02.7350446Z info: downloading installer
2021-12-07T03:32:04.7791968Z info: profile set to 'default'
2021-12-07T03:32:04.7814110Z info: default host triple is x86_64-unknown-linux-gnu
2021-12-07T03:32:04.7827941Z info: skipping toolchain installation
2021-12-07T03:32:04.7838542Z 
2021-12-07T03:32:04.7845449Z 
2021-12-07T03:32:04.7847648Z Rust is installed now. Great!
2021-12-07T03:32:04.7852166Z 
2021-12-07T03:32:04.7854159Z To get started you may need to restart your current shell.
2021-12-07T03:32:04.7885600Z This would reload your PATH environment variable to include
2021-12-07T03:32:04.7888291Z Cargo's bin directory ($HOME/.cargo/bin).
2021-12-07T03:32:04.7889852Z 
2021-12-07T03:32:04.7891069Z To configure your current shell, run:
2021-12-07T03:32:04.7892227Z source $HOME/.cargo/env
2021-12-07T03:32:04.7903454Z ##[error]Unable to process command '::add-path::/home/ghactions/.cargo/bin' successfully.
2021-12-07T03:32:04.7920521Z ##[error]The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/
2021-12-07T03:32:05.2771794Z Installing toolchain with components and targets: stable -- linux --  -- 
2021-12-07T03:32:05.2777634Z [command]/home/ghactions/.cargo/bin/rustup toolchain install stable --profile minimal --allow-downgrade
2021-12-07T03:32:05.2825293Z info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
2021-12-07T03:32:05.8396842Z info: latest update on 2021-12-02, rust version 1.57.0 (f1edd0429 2021-11-29)
2021-12-07T03:32:05.8400187Z info: downloading component 'cargo'
2021-12-07T03:32:06.5955069Z info: downloading component 'rust-std'
2021-12-07T03:32:09.1956247Z info: downloading component 'rustc'
2021-12-07T03:32:14.6039407Z info: installing component 'cargo'
2021-12-07T03:32:15.3460818Z info: installing component 'rust-std'
2021-12-07T03:32:17.9211257Z info: installing component 'rustc'
2021-12-07T03:32:23.0377666Z 
2021-12-07T03:32:23.0521404Z   stable-x86_64-unknown-linux-gnu installed - rustc 1.57.0 (f1edd0429 2021-11-29)
2021-12-07T03:32:23.0522647Z 
2021-12-07T03:32:23.0878768Z info: default toolchain set to 'stable-x86_64-unknown-linux-gnu'
2021-12-07T03:32:23.0885988Z info: checking for self-updates
2021-12-07T03:32:23.1339840Z Setting the default toolchain: stable
2021-12-07T03:32:23.1343964Z [command]/home/ghactions/.cargo/bin/rustup default stable
2021-12-07T03:32:23.1391753Z info: using existing install for 'stable-x86_64-unknown-linux-gnu'
2021-12-07T03:32:23.1711115Z info: default toolchain set to 'stable-x86_64-unknown-linux-gnu'
2021-12-07T03:32:23.1712082Z 
2021-12-07T03:32:23.1855712Z   stable-x86_64-unknown-linux-gnu unchanged - rustc 1.57.0 (f1edd0429 2021-11-29)
2021-12-07T03:32:23.1856756Z 
2021-12-07T03:32:23.7328678Z [command]/usr/bin/tar --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/ghactions/actions-runner/_work/fast-hampath/fast-hampath --files-from manifest.txt
2021-12-07T03:33:11.0219002Z Cache saved successfully
2021-12-07T03:33:11.0383823Z Post job cleanup.
2021-12-07T03:33:11.1697119Z [command]/usr/bin/git version
2021-12-07T03:33:11.1758083Z git version 2.27.0
2021-12-07T03:33:11.1799220Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2021-12-07T03:33:11.1846511Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2021-12-07T03:33:11.2251710Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2021-12-07T03:33:11.2288738Z http.https://github.com/.extraheader
2021-12-07T03:33:11.2297676Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2021-12-07T03:33:11.2342274Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2021-12-07T03:33:11.2794211Z Cleaning up orphan processes
```
</details>

#### Possible Solutions

I could set `ACTIONS_ALLOW_UNSECURE_COMMANDS` as suggested by the error message, but the ultimate goal of this runner is to run regression benchmarks for starship, which means that arbitrary PRs might be run against it. Accordingly, I would like to keep the attack surface as low as I can and I don't know what else that envar enables.

Following several levels of links from the blog post in the error message shows docs that [suggest editing PATH by writing output to the `$GITHUB_PATH` file](https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#adding-a-system-path), but I haven't tried this myself and I don't know how I could (still relatively new to the Actions infrastructure--I'm more used to GitLab CI).

#### System-info

- OS: Rocky Linux 8.5
- Version: Latest action, with runner version 2.285.0
- Rust Version: Stable

**Comment by ATiltedTree** [comment]
Should now be fixed with v1.0.5!

**Comment by chipbuster** [comment]
Works like a charm. Thanks for the quick fix!
